community/kernel/src/main/java/org/neo4j/kernel/impl/api/StateHandlingStatementOperations.java

```diff
@@ -362,46 +362,33 @@ public boolean nodeExists( KernelStatement state, long nodeId )
     @Override
     public boolean relationshipExists( KernelStatement state, long relId )
     {
-        try ( RelationshipCursor cursor = relationshipCursor( state, relId ) )
+        try ( Cursor<RelationshipItem> cursor = relationshipCursor( state, relId ) )
         {
             return cursor.next();
         }
     }
 
     @Override
-    public boolean nodeHasLabel( KernelStatement state, long nodeId, int labelId ) throws EntityNotFoundException
+    public boolean nodeHasLabel( KernelStatement state, NodeItem node, int labelId )
     {
-        try ( NodeCursor nodeCursor = nodeCursor( state, nodeId ) )
+        try ( Cursor<LabelItem> labelCursor = node.label( labelId ) )
         {
-            if ( nodeCursor.next() )
-            {
-                try ( LabelCursor labelCursor = nodeCursor.labels() )
-                {
-                    return labelCursor.seek( labelId );
-                }
-            }
-            throw new EntityNotFoundException( EntityType.NODE, nodeId );
+            return labelCursor.next();
         }
     }
 
     @Override
-    public PrimitiveIntIterator nodeGetLabels( KernelStatement state, long nodeId ) throws EntityNotFoundException
+    public PrimitiveIntIterator nodeGetLabels( KernelStatement state, NodeItem node )
     {
-        return nodeGetLabels( state, state.getStoreStatement(), nodeId );
+        return nodeGetLabels( state, state.getStoreStatement(), node );
     }
 
     @Override
-    public PrimitiveIntIterator nodeGetLabels( TxStateHolder txStateHolder, StoreStatement storeStatement, long nodeId )
-            throws EntityNotFoundException
+    public PrimitiveIntIterator nodeGetLabels( TxStateHolder txStateHolder,
+            StoreStatement storeStatement,
+            NodeItem node )
     {
-        try ( NodeCursor nodeCursor = nodeCursor( txStateHolder, storeStatement, nodeId ) )
-        {
-            if ( nodeCursor.next() )
-            {
-                return Cursors.intIterator( nodeCursor.labels(), LabelCursor.GET_LABEL );
-            }
-            throw new EntityNotFoundException( EntityType.NODE, nodeId );
-        }
+        return Cursors.intIterator( node.labels(), LabelItem.GET_LABEL );
```

Comment:
Actually, the logical place would be Primitive.intIterator, but that class cannot see org.neo4j.graphdb.Resource. We would need to refactor package so that this works. Good thing to do for 3.0 perhaps.